pymatviz/widgets/matterviz.py

```
@@ -269,3 +273,130 @@ def to_dict(self) -> dict[str, Any]:
+                        f"supported: {sorted(valid_fmts)}"
+                    )
+                resolved_fmt = ext
+            else:
+                resolved_fmt = "png"
+
+        # Ensure ESM/CSS assets are loaded
+        cls = type(self)
+        cls._set_class_assets()
+        assets = cls._asset_cache["default"]
+        if not isinstance(assets, tuple):
+            raise TypeError("Widget assets not loaded correctly")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)